tutorial.md

```diff
@@ -0,0 +1,156 @@
+---
+Author: Dan Gunter
+Revised: 4 December 2022
+---
+
+# IDAES Examples Tutorial
+
+For developers.
+
+Assumptions: user = 'username'
+
+## Setup
+
+### Fork and clone the repository
+
+
+Go to https://github.com/IDAES/examples and fork the repository.
+
+Start a shell and change to directory where you will clone the repository.
+```
+git clone  https://github.com/username/examples
+cd examples
+git remote add upstream  https://github.com/username/examples
+```
+
+### Install in a Conda environment
+
+Continue in the directory where you cloned the repository.
+Create a conda environment and install examples (and dependencies, including idaes-pse).
+
+```
+conda create -y -n idaes-examples python=3
+conda activate idaes-examples
+pip install -e .[dev]
```

Comment:
but I'm trying to get away from requirements entirely, and just use pyproject.toml for everything

Reply:
OK, that makes sense. A bit of a pedantic comment, but technically the discussion of whether to use `requirements.txt` or not is independent from the adoption of `pyproject.toml`:

- `pyproject.toml` is the (general) specification for any PEP-517-compliant tool, and it is a direct replacement of `setup.py`
- `requirements.txt` is a format specific to pip, i.e. the particular "build front-end" tool that we use

Therefore we could continue to use `requirements.txt` as basically a snapshot of `pip install` command-line options, while having `pyproject.toml` (and no `setup.py`) since pip supports PEP-517. If we wanted to go for another PEP-517-compliant tool, i.e. Poetry, then we'd need `pyproject.toml` but no `requirements.txt`.